lib/brainstem/dsl/configuration.rb

```diff
@@ -11,8 +11,13 @@ class Configuration
       # @params [Object] parent_configuration The parent configuration object
       #   which the new configuration object should use as a base.
       def initialize(parent_configuration = nil)
-        @parent_configuration = parent_configuration || ActiveSupport::HashWithIndifferentAccess.new
-        @storage = ActiveSupport::HashWithIndifferentAccess.new
+        @parent_configuration  = parent_configuration || ActiveSupport::HashWithIndifferentAccess.new
+        @storage               = ActiveSupport::HashWithIndifferentAccess.new
+
+        self.nonheritable_keys =
+          parent_configuration.respond_to?(:nonheritable_keys) &&
+          parent_configuration.nonheritable_keys ||
```

Comment:
I feel like I maybe don't quite understand the question here.

If the parent changes this list, child classes would also reflect that change. But unless I'm grossly misunderstanding the way that configuration is likely to be used, an annotation of non-heritability is more than likely to be limited to this library's code, rather than in any end-user code which might lead to a breakdown in functionality. As of right now, the use of `nonheritable!` is constrained to the `title`, `description`, and `nodoc!` annotations, all of which (to my mind) should be *inheritably* non-heritable.

Basically, if we suddenly decide that `:associations` is non-heritable, or decide that `:title` should be heritable, we're going to have to make changes, but the trade-off to this is that we don't constantly have to be re-adding these annotations to sub-configurations, especially in the action context where we're manually instantiating `Configuration` objects, and where the non-heritability of `title` still remains important.

But then again, I could very well be misunderstanding the question. `Configuration` so easily twists my head about. It's the Inception of data structures.